File: engine/transpile/src/test/resources/golden/SYK006/SYK3_在庫抽出レコード.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK006 のデータ部レコード SYK3-在庫抽出レコード を逐語対訳した自動生成コード。 */
public final class SYK3_在庫抽出レコード {

    /** 01 SYK3-在庫抽出レコード 総バイト長 29。 */
    public static final int LENGTH = 29;

    private final byte[] data;

    public SYK3_在庫抽出レコード() {
        this.data = new byte[29];
    }

    public SYK3_在庫抽出レコード(byte[] source) {
        this.data = Arrays.copyOf(source, 29);
    }

    public byte[] data() {
        return data;
    }

    // SYK3-商品コード
    public String get_SYK3_商品コード() {
        return CobolRuntime.decodeAlphanumeric(data, 0, 8);
    }
    public void set_SYK3_商品コード(String value) {
        CobolRuntime.encodeAlphanumeric(data, 0, 8, value);
    }

    // SYK3-倉庫コード
    public String get_SYK3_倉庫コード() {
        return CobolRuntime.decodeAlphanumeric(data, 8, 4);
    }
    public void set_SYK3_倉庫コード(String value) {
        CobolRuntime.encodeAlphanumeric(data, 8, 4, value);
    }

    // SYK3-在庫数量
    public long get_SYK3_在庫数量() {
        return CobolRuntime.decodePacked(data, 12, 4);
    }
    public void set_SYK3_在庫数量(long value) {
        CobolRuntime.encodePacked(data, 12, 4, true, value);
    }

    // SYK3-引当可能数量
    public long get_SYK3_引当可能数量() {
        return CobolRuntime.decodePacked(data, 16, 4);
    }
    public void set_SYK3_引当可能数量(long value) {
        CobolRuntime.encodePacked(data, 16, 4, true, value);
    }

    // SYK3-更新区分
    public String get_SYK3_更新区分() {
        return CobolRuntime.decodeAlphanumeric(data, 20, 1);
    }
    public void set_SYK3_更新区分(String value) {
        CobolRuntime.encodeAlphanumeric(data, 20, 1, value);
    }

    public boolean is_SYK3_増加() {
        return get_SYK3_更新区分().equals("1");
    }

    public boolean is_SYK3_減少() {
        return get_SYK3_更新区分().equals("2");
    }

    // SYK3-処理日
    public long get_SYK3_処理日() {
        return CobolRuntime.decodeZoned(data, 21, 8);
    }
    public void set_SYK3_処理日(long value) {
        CobolRuntime.encodeZoned(data, 21, 8, false, value);
    }
}

File: src/engine/transpile/src/test/resources/golden/SYK007/SYK3_在庫抽出レコード.py
"""SYK007 のデータ部レコード SYK3-在庫抽出レコード を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class SYK3_在庫抽出レコード:
    """01 SYK3-在庫抽出レコード 総バイト長 29。"""

    LENGTH = 29

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(29)
        else:
            self._data = bytearray(data)
            if len(self._data) < 29:
                self._data.extend(b"\x00" * (29 - len(self._data)))

    @property
    def data(self):
        return self._data

    # SYK3-商品コード
    def get_SYK3_商品コード(self):
        return decode_alphanumeric(self._data, 0, 8)
    def set_SYK3_商品コード(self, value):
        encode_alphanumeric(self._data, 0, 8, value)

    # SYK3-倉庫コード
    def get_SYK3_倉庫コード(self):
        return decode_alphanumeric(self._data, 8, 4)
    def set_SYK3_倉庫コード(self, value):
        encode_alphanumeric(self._data, 8, 4, value)

    # SYK3-在庫数量
    def get_SYK3_在庫数量(self):
        return decode_packed(self._data, 12, 4)
    def set_SYK3_在庫数量(self, value):
        encode_packed(self._data, 12, 4, True, value)

    # SYK3-引当可能数量
    def get_SYK3_引当可能数量(self):
        return decode_packed(self._data, 16, 4)
    def set_SYK3_引当可能数量(self, value):
        encode_packed(self._data, 16, 4, True, value)

    # SYK3-更新区分
    def get_SYK3_更新区分(self):
        return decode_alphanumeric(self._data, 20, 1)
    def set_SYK3_更新区分(self, value):
        encode_alphanumeric(self._data, 20, 1, value)

    def is_SYK3_増加(self):
        return self.get_SYK3_更新区分() == "1"

    def is_SYK3_減少(self):
        return self.get_SYK3_更新区分() == "2"

    # SYK3-処理日
    def get_SYK3_処理日(self):
        return decode_zoned(self._data, 21, 8)
    def set_SYK3_処理日(self, value):
        encode_zoned(self._data, 21, 8, False, value)
